src/test/java/school/hei/haapi/unit/PaymentServiceUnitTest.java

```diff
@@ -0,0 +1,153 @@
+    associatedFee.setTotalAmount(5000);
+    associatedFee.setStatus(LATE);
+    associatedFee.setCreationDatetime(Instant.parse("2022-12-08T08:25:24.00Z"));
+    associatedFee.setDueDatetime(Instant.parse("2023-02-08T08:30:24.00Z"));
+    associatedFee.setUpdatedAt(Instant.parse("2021-12-09T08:25:24.00Z"));
+    return associatedFee;
+  }
+
+  public static User student1() {
+    User student1 = new User();
+    student1.setId("student1_id");
+    student1.setFirstName("Ryan");
+    student1.setLastName("Andria");
+    student1.setEmail("test+ryan@hei.school");
+    student1.setRef("STD21001");
+    student1.setStatus(ENABLED);
+    student1.setSex(M);
+    student1.setBirthDate(LocalDate.parse("2000-01-01"));
+    student1.setEntranceDatetime(Instant.now());
+    student1.setPhone("0123456789");
+    student1.setAddress("Example Address");
+    return student1;
+  }
+
+  public static User student2() {
+    User student2 = new User();
+    student2.setId("student2_id");
+    student2.setFirstName("Two");
+    student2.setLastName("Student");
+    student2.setEmail("test+student2@hei.school");
+    student2.setRef("STD21002");
+    student2.setStatus(ENABLED);
+    student2.setSex(F);
+    student2.setBirthDate(LocalDate.parse("2000-01-02"));
+    student2.setEntranceDatetime(Instant.now());
+    student2.setPhone("0322411124");
+    student2.setAddress("Adr 2");
+    return student2;
+  }
+
+  public static User student3() {
+    User student2 = new User();
+    student2.setId("student3_id");
+    student2.setFirstName("Three");
+    student2.setLastName("Student");
+    student2.setEmail("test+student3@hei.school");
+    student2.setRef("STD21003");
+    student2.setStatus(ENABLED);
+    student2.setSex(F);
+    student2.setBirthDate(LocalDate.parse("2000-01-02"));
+    student2.setEntranceDatetime(Instant.now());
+    student2.setPhone("0322411124");
+    student2.setAddress("Adr 2");
+    return student2;
+  }
+
+  @Test
+  @DirtiesContext
+  void compute_remaining_amount_ok() {
+    Fee associatedFee1 = student1UnpaidFee1();
+
+    paymentService.computeRemainingAmount(associatedFee1.getId(), 5000);
+
+    Fee updatedFee = feeService.getById(associatedFee1.getId());
+
+    assertEquals(0, updatedFee.getRemainingAmount());
+    assertEquals(PAID, updatedFee.getStatus());
+  }
+
+  @Test
+  @DirtiesContext
+  void compute_user_status_after_paying_fee_ok() {
+    User userWithUnpaidFees = student2();
+    User userWithoutUnpaidFees = student3();
+
+    paymentService.computeUserStatusAfterPayingFee(userWithUnpaidFees);
+    paymentService.computeUserStatusAfterPayingFee(userWithoutUnpaidFees);
+    User updatedUserWithUnpaidFees = userService.findById(userWithUnpaidFees.getId());
+    User updatedUserWithoutUnpaidFees = userService.findById(userWithoutUnpaidFees.getId());
+
+    assertEquals(SUSPENDED, updatedUserWithUnpaidFees.getStatus());
+    assertEquals(ENABLED, updatedUserWithoutUnpaidFees.getStatus());
```

Comment:
Your test doesn't cover all case. Add the case where the userWithUnpaidFees pay his fees and must be ENABLED